File: Harsh Gupta/Day 8/151. Reverse Words in a String.cpp
class Solution {
public:
    string reverseWords(string s) {
        vector<string> a;
        stringstream ss(s);
        string temp;
        while(ss>>temp){
            a.push_back(temp);
        }
        int i = 0, j= a.size()-1;
        while(i<j){
            temp = a[i];
            a[i] = a[j];
            a[j] = temp;
            i++; j--;
        }
        temp ="";
        j= a.size();
        for(int i = 0; i<j ; i++){
            temp+=a[i];
            if(i == j-1){
                break;
            }
            temp+=" ";
        }
        return temp;
    }
};

File: Harsh Gupta/Day 8/151. Reverse Words in a String.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        s = s.split()
        i = 0
        j = len(s)-1
        while(i<j):
            temp = s[i]
            s[i] = s[j]
            s[j] = temp
            i+=1
            j-=1
        return (' '.join(s))
